rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -187,14 +187,17 @@ local function route_to_alternative_balancer(balancer)
   local header = ngx.var["http_" .. target_header]
   if header then
     if traffic_shaping_policy.headerValue
-       and #traffic_shaping_policy.headerValue > 0 then
+	   and #traffic_shaping_policy.headerValue > 0 then
       if traffic_shaping_policy.headerValue == header then
         return true
       end
-
+    elseif traffic_shaping_policy.headerPattern
+       and #traffic_shaping_policy.headerPattern > 0 then
+      if ngx.re.match(header, traffic_shaping_policy.headerPattern) then
```

Comment:
If you are using `ngx.re.match` here please make sure you handle potential error too, see the API https://github.com/openresty/lua-nginx-module#ngxrematch

Also I'd consider adding `jo` option for compiling the regex and caching it per worker process.

Reply:
done